src/App.tsx

```diff
@@ -1,87 +1,96 @@
+}
+const store = new Store();
 
 function App() {
-  const [tasks, setTasks] = useState<Task[]>([]);
-  const [hideCompleted, setHideCompleted] = useState(false);
+  return <Todo store={store} />
 
+}
+const Todo: React.FC<{ store: Store }> = observer(({ store }) => {
   useEffect(() => {
-    fetchTasks(hideCompleted).then(setTasks);
-  }, [hideCompleted]);
-
-  const addTask = () => {
-    setTasks([...tasks, new Task()])
-  };
-  const setAll = async (completed: boolean) => {
-    await TasksController.setAll(completed);
-    setTasks(await fetchTasks(hideCompleted));
-  }
-
+    store.loadTasks()
+  }, [store.hideCompleted]);
   return (
     <div>
       <input
         type="checkbox"
-        checked={hideCompleted}
-        onChange={e => setHideCompleted(e.target.checked)} /> Hide Completed
+        checked={store.hideCompleted}
+        onChange={action(e => store.hideCompleted = e.target.checked)} /> Hide Completed
       <main>
-        {tasks.map(task => {
-          const handleChange = (values: Partial<Task>) => {
-            setTasks(tasks.map(t => t === task ? { ...task, ...values } : t));
-          };
-
-          const saveTask = async () => {
-            try {
-              const savedTask = await taskRepo.save(task);
-              setTasks(tasks.map(t => t === task ? savedTask : t));
-            } catch (error: any) {
-              alert(error.message);
-            }
-          }
-
-
-          const deleteTask = async () => {
-            await taskRepo.delete(task);
-            setTasks(tasks.filter(t => t !== task));
-          };
-
+        {store.tasks.map(task => {
           return (
             <div key={task.id}>
               <input type="checkbox"
                 checked={task.completed}
-                onChange={e => handleChange({ completed: e.target.checked })} />
+                onChange={action(e => {
+                  const prev = task.completed;
+                  task.completed = e.target.checked;
+                })} />
               <input
                 value={task.title}
-                onChange={e => handleChange({ title: e.target.value })} />
-              <button onClick={saveTask}>Save</button>
-              <button onClick={deleteTask}>Delete</button>
+                onChange={action(e => task.title = e.target.value)} />
+              <button onClick={() => store.saveTask(task)}>Save</button>
+              <button onClick={() => store.deleteTask(task)}>Delete</button>
             </div>
           );
         })}
 
       </main>
-      <button onClick={addTask}>Add Task</button>
+      <button onClick={() => store.addTask()}>Add Task</button>
       <div>
-        <button onClick={() => setAll(true)}>Set all as completed</button>
-        <button onClick={() => setAll(false)}>Set all as uncompleted</button>
+        <button onClick={() => store.setAll(true)}>Set all as completed</button>
+        <button onClick={() => store.setAll(false)}>Set all as uncompleted</button>
```

Comment:
Maintainability-wise, `setAll(boolean)` is terrible cause it doesn't tell the reader what does the `setAll` set. This has a double cause: the non descriptive name + boolean argument. I suggest naming the action descriptively and either passing an `isCompleted: boolean` object as an argument - or, better - creating two named actions which don't accept any arguments: `setAllCompleted(): void` and `setAllUncompleted(): void`.